src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate before = parseString(dateFrom);
+        LocalDate after = parseString(dateTo);
+        StringBuilder sb = new StringBuilder();
+        sb.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            sb.append(System.lineSeparator())
+                    .append(name).append(" - ")
+                    .append(streamCalculator(data, after, before, name));
+        }
+        return sb.toString();
+    }
+
+    private Integer streamCalculator(String[] data, LocalDate after,
```

Comment:
```suggestion
    private Integer calculateSalary(String[] data, LocalDate after,
```
method names should be verbs